src/engine/QueryExecutionTree.cpp

```diff
@@ -399,38 +355,66 @@ nlohmann::json QueryExecutionTree::writeQLeverJsonTable(
   return json;
 }
 
+// _____________________________________________________________________________
+
+vector<std::optional<pair<size_t, ResultTable::ResultType>>>
+QueryExecutionTree::buildValidIndices(const vector<string>& selectVars,
+                                      const ResultTable& resultTable) const {
+  vector<std::optional<pair<size_t, ResultTable::ResultType>>> validIndices;
+  for (auto var : selectVars) {
+    constexpr std::string_view prefix = "TEXT(";
+    constexpr size_t prefixLength = prefix.length();
+    if (var.starts_with(prefix)) {
+      var = var.substr(prefixLength, var.rfind(')') - prefixLength);
+    }
+    auto it = getVariableColumns().find(var);
+    if (it != getVariableColumns().end()) {
+      validIndices.emplace_back(pair<size_t, ResultTable::ResultType>(
+          it->second, resultTable.getResultType(it->second)));
+    } else {
+      validIndices.emplace_back(std::nullopt);
+    }
+  }
+  return validIndices;
+}
+
```

Comment:
This function already exists and is called `selectedVariablesToColumnIndices`, 
and returns a proper type, and lives in line 137.

Reply:
But of course you can apply your improvements there (the prefix size, etc.)